presto-spi/src/test/java/io/prestosql/spi/testing/InterfaceTestUtils.java

```diff
@@ -47,6 +47,10 @@ private InterfaceTestUtils() {}
             if (method.getDeclaringClass() == Object.class) {
                 continue;
             }
+            if (method.getAnnotation(Deprecated.class) != null) {
+                // deprecated methods in interfaces exist for backwards compatibility only; they don't have to be implemented
```

Comment:
This is, in general, not safe. It's better to have explicit per-test exclusions

Reply:
This became de facto a 'pattern' in Presto code, other interfaces adhere to it as well.

Also if we use the solution from my next comment, this reflection-based check test becomes unnecessary.

Reply:
done, but I don't like this solution

we should start using something else for backwards compatibility ASAP, for example something like `XxxxResponse xxxx(XxxxRequest request)`, even if it may look exaggerated in the beginning